default.nix

```diff
@@ -122,11 +125,33 @@ in {
           default = [ ];
           type = with lib.types; listOf (either str attrs);
         };
+
         options.bridge = lib.mkOption {
           description = "Add this device to this bridge";
           default = null;
-          type = with lib.types; nullOr str;
+          type = with lib.types; nullOr (coercedTo str (name: { inherit name; }) (submodule {
+            options.name = lib.mkOption {
+              description = "Name of the bridge";
+              type = lib.types.str;
+            };
+            options.vlans = lib.mkOption {
```

Comment:
This option doesn't belong here, `bridge` belongs to interface config because it relates to the interface (what bridge the interface should be added to), but `vlans` specifies what should be done to the bridge, not the interface. Please add a separate `router.bridges` submodule, similar to `router.vethPairs`.